### PR TITLE
Add CRD openapi definitions

### PR DESCRIPTION
**Pull request by soltysh**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1688638

/assign @deads2k @smarterclayton @sttts

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -15,6 +15,11 @@ required = [ "k8s.io/code-generator/cmd/client-gen", "k8s.io/gengo/types" ]
   non-go = false
   unused-packages = false
 
+[[prune.project]]
+  name = "github.com/openshift/library-go"
+  non-go = false
```
Comment (abhinavdahiya) [inline Gopkg.toml]: update the commit message with reason for prune exception

Comment (soltysh) [reply Gopkg.toml]: Done.

**Comment by soltysh** [comment]
@abhinavdahiya ptal

**Comment by abhinavdahiya** [comment]
we can't move forawrd on this as a lot of operators are failing the validation to update the clusteroperator:
example monitoring:
```
error occurred while setting status to in progress: ClusterOperator.config.openshift.io "monitoring" is invalid: []: Invalid value: map[string]interface {}{"kind":"ClusterOperator", "metadata":map[string]interface {}{"uid":"96475849-66d6-11e9-a6c8-12da682b8aa6", "generation":1, "creationTimestamp":"2019-04-24T21:19:04Z", "resourceVersion":"8619", "name":"monitoring"}, "status":map[string]interface {}{"versions":interface {}(nil), "relatedObjects":interface {}(nil), "extension":interface {}(nil), "conditions":[]interface {}{map[string]interface {}{"lastTransitionTime":"2019-04-24T21:19:04Z", "type":"Failing", "status":"Unknown"}, map[string]interface {}{"type":"Available", "status":"Unknown", "lastTransitionTime":"2019-04-24T21:19:04Z"}, map[string]interface {}{"type":"Progressing", "status":"True", "lastTransitionTime":"2019-04-24T21:19:04Z", "message":"Rolling out the stack."}}}, "spec":map[string]interface {}{}, "apiVersion":"config.openshift.io/v1"}: validation failure list:
status.relatedObjects in body must be of type array: "null"
status.versions in body must be of type array: "null"
```

**Comment by soltysh** [comment]
Monitoring is using old version of openshift/api which is missing the necessary tags, I've opened https://github.com/openshift/cluster-monitoring-operator/pull/338 to address that.

**Comment by soltysh** [comment]
Actually the PR to look for is https://github.com/openshift/cluster-monitoring-operator/pull/333

**Comment by soltysh** [comment]
I'm checking other pods/operator for errors as such.

**Comment by soltysh** [comment]
Another to blame is cluster-storage-operator, I'm just talking with them about the PR.

**Comment by soltysh** [comment]
Also operator-marketplace, also talking with them about bump.

**Comment by soltysh** [comment]
Prereqs merged, let's
/retest

**Comment by soltysh** [comment]
Rebased, let's see the current results.

**Comment by soltysh** [comment]
Further failures:
- `status.conditions.lastTransitionTime in body must be of type string: \"null\"` in OLM - fix in https://github.com/operator-framework/operator-lifecycle-manager/pull/830

**Pull request merged**